c_src/ex_dtls/native.c

```diff
@@ -102,8 +102,10 @@ UNIFEX_TERM do_handshake(UnifexEnv *env, State *state, UnifexPayload *payload) {
     case SSL_ERROR_WANT_READ:
       DEBUG("SSL WANT READ");
       state->SSL_error = SSL_ERROR_WANT_READ;
-      // break and wait for data from remote host. It will come in feed() function.
-      UnifexPayload *payload = (UnifexPayload *)unifex_payload_alloc(env, UNIFEX_PAYLOAD_BINARY, dyn_buff->data_size);
+      // break and wait for data from remote host. It will come in feed()
+      // function.
+      UnifexPayload *payload = (UnifexPayload *)unifex_payload_alloc(
```

Comment:
whenever you do unifex_payload_alloc, unless you want to store it on your own, you should do unifex_payload_release like here: https://github.com/membraneframework/membrane_mp3_lame_plugin/blob/154c43594972c5cc27df65d05d6fb1aaf9ffc956/c_src/membrane_mp3_lame_plugin/encoder.c#L121 
Otherwise, you'll have a memory leak. The same applies to unifex_alloc_state. Please check all the occurrences